source/tests/unit/ni_fake_service_tests.cpp

```diff
@@ -1198,6 +1199,81 @@ TEST(NiFakeServiceTests, NiFakeService_GetAttributeViString_CallsGetAttributeViS
   EXPECT_THAT(response.attribute_value(), ElementsAreArray(attribute_char_array, expected_size));
 }
 
+TEST(NiFakeServiceTests, NiFakeService_GetViUInt8_CallsGetViUInt8)
+{
+  nidevice_grpc::SessionRepository session_repository;
+  std::uint32_t session_id = create_session(session_repository, kTestViSession);
+  NiFakeMockLibrary library;
+  nifake_grpc::NiFakeService service(&library, &session_repository);
+  ViUInt8 a_ViUInt8_number = 255;
+  EXPECT_CALL(library, GetViUInt8(kTestViSession, _))
+      .WillOnce(DoAll(SetArgPointee<1>(a_ViUInt8_number), Return(kDriverSuccess)));
+
+  ::grpc::ServerContext context;
+  nifake_grpc::GetViUInt8Request request;
+  request.mutable_vi()->set_id(session_id);
+  nifake_grpc::GetViUInt8Response response;
+  ::grpc::Status status = service.GetViUInt8(&context, &request, &response);
+
+  EXPECT_TRUE(status.ok());
+  EXPECT_EQ(kDriverSuccess, response.status());
+  EXPECT_EQ(a_ViUInt8_number, response.a_uint8_number());
+}
+
+TEST(NiFakeServiceTests, NiFakeService_ViUInt8ArrayInputFunction_CallsViUInt8ArrayInputFunction)
+{
+  nidevice_grpc::SessionRepository session_repository;
+  std::uint32_t session_id = create_session(session_repository, kTestViSession);
+  NiFakeMockLibrary library;
+  nifake_grpc::NiFakeService service(&library, &session_repository);
+  ViInt32 number_of_elements = 3;
+  ViUInt8 expected_array[] = {0, 127, 255};
+  EXPECT_CALL(library, ViUInt8ArrayInputFunction(kTestViSession, number_of_elements, _))
+    .With(Args<2, 1>(ElementsAreArray(expected_array)))
+    .WillOnce(Return(kDriverSuccess));
+
+  ::grpc::ServerContext context;
+  nifake_grpc::ViUInt8ArrayInputFunctionRequest request;
+  request.mutable_vi()->set_id(session_id);
+  request.set_number_of_elements(number_of_elements);
+  std::string input_array;
+  input_array.push_back(0);
+  input_array.push_back(127);
+  input_array.push_back(255);
+  request.set_an_array(input_array);
+  nifake_grpc::ViUInt8ArrayInputFunctionResponse response;
+  ::grpc::Status status = service.ViUInt8ArrayInputFunction(&context, &request, &response);
+
+  EXPECT_TRUE(status.ok());
+  EXPECT_EQ(kDriverSuccess, response.status());
+}
+
+TEST(NiFakeServiceTests, NiFakeService_ViUInt8ArrayOutputFunction_CallsViUInt8ArrayOutputFunction)
+{
+  nidevice_grpc::SessionRepository session_repository;
+  std::uint32_t session_id = create_session(session_repository, kTestViSession);
+  NiFakeMockLibrary library;
+  nifake_grpc::NiFakeService service(&library, &session_repository);
+  ViInt32 number_of_elements = 3;
+  ViUInt8 an_array[] = {0, 127, 255};
```

Comment:
Good call using values from throughout UInt8 range. Jonathan had suggested in another PR to specify upper value in hex representation so it is clear that we are using max 8 bit value, like `0xFF`. Make that change here as well.